server/service/osquery.go

```diff
@@ -1830,6 +1834,180 @@ func (svc *Service) processSoftwareForNewlyFailingPolicies(
 	return nil
 }
 
+func (svc *Service) processScriptsForNewlyFailingPolicies(
```

Comment:
The initial logic to calculate policies that are failing for first time or go from pass to fail is similar to the installer case, we should try to re-use if possible. Am ok with duplicating if it ends up with too many arguments to re-use.

Reply:
Am ok with this actually after seeing the implementation :)